### PR TITLE
Detect links and render them as HTML links in repo description

### LINKED ISSUE #943: Detect URLs in project description
...And render them as hyperlinks. As on GitHub.

### PR DESCRIPTION
**Pull request by lefou**
This fixes #943

### CODE REVIEW

File: src/main/scala/gitbucket/core/view/helpers.scala [hunk 1]
```diff
@@ -275,4 +275,10 @@ object helpers extends AvatarImageProvider with LinkConverter with RequestCache
...
...v|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)/)(?:[^\s()<>{}\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\))+(?:\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’])|(?:(?<!@)[a-z0-9]+(?:[.\-][a-z0-9]+)*[.](?:com|net|org|edu|gov|mil|aero|asia|biz|cat|coop|info|int|jobs|mobi|museum|name|post|pro|tel|travel|xxx|ac|ad|ae|af|ag|ai|al|am|an|ao|aq|ar|as|at|au|aw|ax|az|ba|bb|bd|be|bf|bg|bh|bi|bj|bm|bn|bo|br|bs|bt|bv|bw|by|bz|ca|cc|cd|cf|cg|ch|ci|ck|cl|cm|cn|co|cr|cs|cu|cv|cx|cy|cz|dd|de|dj|dk|dm|do|dz|ec|ee|eg|eh|er|es|et|eu|fi|fj|fk|fm|fo|fr|ga|gb|gd|ge|gf|gg|gh|gi|gl|gm|gn|gp|gq|gr|gs|gt|gu|gw|gy|hk|hm|hn|hr|ht|hu|id|ie|il|im|in|io|iq|ir|is|it|je|jm|jo|jp|ke|kg|kh|ki|km|kn|kp|kr|kw|ky|kz|la|lb|lc|li|lk|lr|ls|lt|lu|lv|ly|ma|mc|md|me|mg|mh|mk|ml|mm|mn|mo|mp|mq|mr|ms|mt|mu|mv|mw|mx|my|mz|na|nc|ne|nf|ng|ni|nl|no|np|nr|nu|nz|om|pa|pe|pf|pg|ph|pk|pl|pm|pn|pr|ps|pt|pw|py|qa|re|ro|rs|ru|rw|sa|sb|sc|sd|se|sg|sh|si|sj|Ja|sk|sl|sm|sn|so|sr|ss|st|su|sv|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)\b/?(?!@)))""".r
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/view/helpers.scala]: This variable should be private. And does it really need an exact regular expression like this?
In markedj which is a new markdown processor for GitBucket, we are using a simpler regular expression:
https://github.com/gitbucket/markedj/blob/master/src/main/java/io/github/gitbucket/markedj/Grammer.java#L92

Comment (lefou) [inline src/main/scala/gitbucket/core/view/helpers.scala]: Agreed to both points. About the regex: I was following a discussion and the history of the regex (for the record: here is the entry point: http://stackoverflow.com/a/4390768/1771641) and had mixed feeling about this explicit and at the same time incompile TLD list. I guess I'll update this PR with the more generic regex I had used initally.

Comment (lefou) [inline src/main/scala/gitbucket/core/view/helpers.scala]: Do I read you correctly, you are in favour to use the same regex as in `markedj`?

Comment (takezoe) [inline src/main/scala/gitbucket/core/view/helpers.scala]: No, it's just my impression :-)

**Pull request merged**